PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs

```diff
@@ -10,7 +13,22 @@ class HighlightTextEnabledHandler : EnabledHandler
     {
         protected override bool GetEnabled(string ribbonId)
         {
-            return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+            try
+            {
+                if (this.GetAddIn().Application.ActiveWindow.Selection.Type == PpSelectionType.ppSelectionText && 
+                    this.GetAddIn().Application.ActiveWindow.Selection.TextRange2.TrimText().Length > 0)
+                {
+                    return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+                }
+                else
+                {
+                    return !this.GetRibbonUi().HighlightTextFragmentsEnabled;
```

Comment:
Is this working as intended? If HighlightTextFragmentsEnabled is false and selectiontextlength is 0, is the button incorrectly enabled?

Reply:
I have changed it;)